osbs/utils.py

```diff
@@ -333,6 +333,40 @@ def utcnow():
 VALID_BUILD_CONFIG_NAME_CHARS = re.compile('[-a-z0-9]')
 
 
+def sanitize_strings_for_openshift(str1, str2='', limit=64, separator='-'):
+    """
+    OpenShift requires labels and build names to be no more than 64 characters and
+    forbids any characters other than alphanumerics and -.
```

Comment:
This isn't strictly true -- dots and underscores are also allowed, but the names cannot begin or end with punctuation.

Reply:
Fixed.